shared/domains/chelonia/chelonia.js

```diff
@@ -1642,50 +1646,110 @@ export default (sbp('sbp/selectors/register', {
   // a general rule, you shouldn't be calling this selector directly unless
   // you're building a utility library or if you have very specific needs. In
   // this case, see if `chelonia/kv/queuedSet` covers your needs.
+  // `data` is allowed to be falsy, in which case a fetch will occur first and
+  // the `onconflict` handler will be called.
   'chelonia/kv/set': async function (contractID: string, key: string, data: Object, {
+    ifMatch,
     innerSigningKeyId,
     encryptionKeyId,
     signingKeyId,
     maxAttempts,
     onconflict
   }: {
+    ifMatch?: string,
     innerSigningKeyId: ?string,
     encryptionKeyId: ?string,
     signingKeyId: string,
     maxAttempts: ?number,
-    onconflict: (contractID: string, key: string, data: Object) => Promise<Object>,
+    onconflict: ChelKvOnConflictCallback,
   }) {
     maxAttempts = maxAttempts ?? 3
+    const url = `${this.config.connectionURL}/kv/${encodeURIComponent(contractID)}/${encodeURIComponent(key)}`
     for (;;) {
-      const serializedData = outputEncryptedOrUnencryptedMessage.call(this, {
-        contractID,
-        innerSigningKeyId,
-        encryptionKeyId,
-        signingKeyId,
-        data,
-        meta: key
-      })
-      const response = await this.config.fetch(`${this.config.connectionURL}/kv/${encodeURIComponent(contractID)}/${encodeURIComponent(key)}`, {
-        headers: new Headers([[
-          'authorization', buildShelterAuthorizationHeader.call(this, contractID)
-        ]]),
-        method: 'POST',
-        body: JSON.stringify(serializedData),
-        signal: this.abortController.signal
-      })
+      let response: Response
+      if (data || typeof onconflict !== 'function') {
```

Comment:
Flowtype mismatch here for `onconflict`: `onconflict` is defined as being a function, yet this condition says it might not be.

Also, is there a situation where both `data` is falsey and `onconflict` is falsey? What happens then?

Reply:
> is there a situation where both data is falsey and onconflict is falsey? What happens then?

In that case a POST will be attempted.

Reply:
> In that case a POST will be attempted.

And what will that POST do? Will it set the data to nothing? Is that the same as deleting data? I feel like we discussed the idea of adding a separate selector for deleting data.... Maybe a comment here about this?

Reply:
Hm, you're right that perhaps we need to specify this better.

No, it will _not_ delete data, but it will allow you to store values such as: `0`, `null`, `''`, `false`.

Reply:
Maybe the check could instead be `if (data === undefined)`?

Reply:
> Maybe the check could instead be if (data === undefined)?

I'm not sure what you mean...

Reply:
I meant that the check could be made stricter and simpler (skipping the `onconflict` part).

Reply:
(I.e., if you want to save `false`, you shouldn't need to set `onconflict` to undefined)